File: sdks/java/flapjacksearch/src/main/java/com/flapjackhq/internal/interceptors/UserAgentInterceptor.java
package com.flapjackhq.internal.interceptors;

import com.flapjackhq.config.FlapjackAgent;
import java.io.IOException;
import javax.annotation.Nonnull;
import okhttp3.Interceptor;
import okhttp3.Response;

public final class UserAgentInterceptor implements Interceptor {

  private final FlapjackAgent agent;

  public UserAgentInterceptor(FlapjackAgent agent) {
    this.agent = agent;
  }

  @Nonnull
  @Override
  public Response intercept(Chain chain) throws IOException {
    okhttp3.Request originalRequest = chain.request();
    okhttp3.Request.Builder builder = originalRequest.newBuilder();
    builder.header("user-agent", agent.toString());
    okhttp3.Request newRequest = builder.build();
    return chain.proceed(newRequest);
  }
}

File: sdks/scala/src/main/scala/flapjacksearch/internal/interceptor/UserAgentInterceptor.scala
package flapjacksearch.internal.interceptor

import flapjacksearch.internal.FlapjackAgent
import okhttp3.{Interceptor, Request, Response}

/** Interceptor that adds the user agent to the request headers.
  *
  * @param agent
  *   user agent
  */
private[flapjacksearch] class UserAgentInterceptor(agent: FlapjackAgent) extends Interceptor {

  override def intercept(chain: Interceptor.Chain): Response = {
    val originalRequest: Request = chain.request()
    val newRequest: Request = originalRequest
      .newBuilder()
      .header("user-agent", agent.toString)
      .build()

    chain.proceed(newRequest)
  }
}
